vertical-pod-autoscaler/pkg/recommender/model/cluster.go

```diff
@@ -509,3 +520,30 @@ func (k aggregateStateKey) Labels() labels.Labels {
 	}
 	return (*k.labelSetMap)[k.labelSetKey]
 }
+
+// ClusterState holds all runtime information about the cluster required for the
+// VPA operations, i.e. configuration of resources (pods, containers,
+// VPA objects), aggregated utilization of compute resources (CPU, memory) and
+// events (container OOMs).
+// All input to the VPA Recommender algorithm lives in this structure.
+type ClusterState interface {
+	StateMapSize() int
+	AddOrUpdatePod(podID PodID, newLabels labels.Set, phase apiv1.PodPhase)
+	GetContainer(containerID ContainerID) *ContainerState
+	DeletePod(podID PodID)
+	AddOrUpdateContainer(containerID ContainerID, request Resources) error
+	AddSample(sample *ContainerUsageSampleWithKey) error
+	RecordOOM(containerID ContainerID, timestamp time.Time, requestedMemory ResourceAmount) error
+	AddOrUpdateVpa(apiObject *vpa_types.VerticalPodAutoscaler, selector labels.Selector) error
+	DeleteVpa(vpaID VpaID) error
+	MakeAggregateStateKey(pod *PodState, containerName string) AggregateStateKey
+	RateLimitedGarbageCollectAggregateCollectionStates(ctx context.Context, now time.Time, controllerFetcher controllerfetcher.ControllerFetcher)
+	RecordRecommendation(vpa *Vpa, now time.Time) error
+	GetMatchingPods(vpa *Vpa) []PodID
+	GetControllerForPodUnderVPA(ctx context.Context, pod *PodState, controllerFetcher controllerfetcher.ControllerFetcher) *controllerfetcher.ControllerKeyWithAPIVersion
+	GetControllingVPA(pod *PodState) *Vpa
+	VPAs() map[VpaID]*Vpa
+	SetObservedVPAs([]*vpa_types.VerticalPodAutoscaler)
+	ObservedVPAs() []*vpa_types.VerticalPodAutoscaler
+	Pods() map[PodID]*PodState
+}
```

Comment:
For some reason I've always thought that interfaces belong near the beginning of the file. Is it worth moving there?
This is just an opinion, so ignore it if it doesn't make sense.

Reply:
You're absolutely right! I'll take care of this later today.

Reply:
done.